hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/staking/StakeInfoHelper.java

```diff
@@ -0,0 +1,128 @@
+
+        final var newStakingInfo = currentStakingInfo.copyBuilder();
+        if (newUnclaimedStakeRewardStart > currentStakeRewardStart) {
+            log.warn(
+                    "Asked to release {} more rewards for node{} (now {}), but only {} was staked",
+                    amount,
+                    nodeId,
+                    newUnclaimedStakeRewardStart,
+                    currentStakeRewardStart);
+            newStakingInfo.unclaimedStakeRewardStart(currentStakeRewardStart);
+        } else {
+            newStakingInfo.unclaimedStakeRewardStart(newUnclaimedStakeRewardStart);
+        }
+
+        stakingInfoStore.put(nodeId, newStakingInfo.build());
+    }
+
+    /**
+     * Awards the stake to the node's stakeToReward or stakeToNotReward depending on the account's decline reward.
+     * If declineReward is true, the stake is awarded to stakeToNotReward, otherwise it is awarded to stakeToReward.
+     * @param nodeId the node's numeric ID
+     * @param account the account stake to be awarded to the node
+     * @param stakingInfoStore the store for the staking info
+     */
+    public void awardStake(
+            @NonNull final Long nodeId,
+            @NonNull final Account account,
+            @NonNull final WritableStakingInfoStore stakingInfoStore) {
+        requireNonNull(nodeId);
+        requireNonNull(account);
+        requireNonNull(stakingInfoStore);
+
+        final var stakeToAward = roundedToHbar(totalStake(account));
+        final var isDeclineReward = account.declineReward();
+
+        final var stakingInfo = stakingInfoStore.get(nodeId);
+        final var copy = stakingInfo.copyBuilder();
+        if (isDeclineReward) {
+            copy.stakeToNotReward(stakingInfo.stakeToNotReward() + stakeToAward);
+        } else {
+            copy.stakeToReward(stakingInfo.stakeToReward() + stakeToAward);
+        }
+        stakingInfoStore.put(nodeId, copy.build());
+    }
+
+    /**
+     * Withdraws the stake from the node's stakeToReward or stakeToNotReward depending on the account's decline reward.
+     * If declineReward is true, the stake is withdrawn from stakeToNotReward, otherwise it is withdrawn from
+     * stakeToReward.
+     * @param nodeId the node's numeric ID
+     * @param account the account 's stake to be withdrawn from node
+     * @param stakingInfoStore the store for the staking info
+     */
+    public void withdrawStake(
+            @NonNull final Long nodeId,
+            @NonNull final Account account,
+            @NonNull final WritableStakingInfoStore stakingInfoStore) {
+        requireNonNull(nodeId);
+        requireNonNull(account);
+        requireNonNull(stakingInfoStore);
+
+        final var stakeToWithdraw = roundedToHbar(totalStake(account));
+        final var isDeclineReward = account.declineReward();
+
+        final var stakingInfo = stakingInfoStore.get(nodeId);
+        final var copy = stakingInfo.copyBuilder();
+        if (isDeclineReward) {
```

Comment:
Also seems absent in mono-service, but wouldn't hurt to have a check here that stake can never become negative.

Reply:
Added a check and also added a warning log